src/Controller/GoogleAuthController.php

```diff
@@ -74,45 +101,100 @@ public static function create(ContainerInterface $container) {
       $container->get('plugin.network.manager'),
       $container->get('social_auth.user_manager'),
       $container->get('social_auth_google.manager'),
-      $container->get('session')
+      $container->get('request_stack'),
+      $container->get('social_auth.social_auth_data_handler'),
+      $container->get('logger.factory')
     );
   }
 
   /**
-   * Redirect to Google Services Authentication page.
+   * Response for path 'user/login/google'.
    *
-   * @return \Zend\Diactoros\Response\RedirectResponse
-   *   Redirection to Google Accounts.
+   * Redirects the user to Google for authentication.
    */
   public function redirectToGoogle() {
-    /* @var \Google_Client $client */
-    $client = $this->networkManager->createInstance('social_auth_google')->getSdk();
-    $client->setScopes(['email', 'profile']);
+    /* @var \League\OAuth2\Client\Provider\Google false $google */
+    $google = $this->networkManager->createInstance('social_auth_google')->getSdk();
+
+    // If google client could not be obtained.
+    if (!$google) {
+      drupal_set_message($this->t('Social Auth Google not configured properly. Contact site administrator.'), 'error');
+      return $this->redirect('user.login');
+    }
 
-    return new RedirectResponse($client->createAuthUrl());
+    // Google service was returned, inject it to $googleManager.
+    $this->googleManager->setClient($google);
+
+    // Generates the URL where the user will be redirected for Google login.
+    // If the user did not have email permission granted on previous attempt,
+    // we use the re-request URL requesting only the email address.
+    $google_login_url = $this->googleManager->getGoogleLoginUrl();
+
+    $state = $this->googleManager->getState();
+
+    $this->dataHandler->set('OAuth2State', $state);
```

Comment:
Keep session variables in lowercase. Sorry if I told you the opposite before (don't remember if I did)
oauth2state